sdk/go/dag/dag.gen.go

```diff
@@ -0,0 +1,240 @@
+// Code generated by dagger. DO NOT EDIT.
+
+package dag
+
+import (
+	"context"
+	"os"
+
+	dagger "dagger.io/dagger"
+)
+
+var client *dagger.Client
+
+func initClient() *dagger.Client {
+	if client == nil {
+		var err error
+		client, err = dagger.Connect(context.Background(), dagger.WithLogOutput(os.Stdout))
+		if err != nil {
+			panic(err)
+		}
+	}
+	return client
+}
```

Comment:
question: should the client from here ever be closed?

Reply:
Yeah.. it's actually occasionally important to explicitly close it for things like upstream remote cache support (which exports the cache once the client is closed). It's not great that it's up to the user to explicitly close a global resource, but I wouldn't consider this a blocker. That whole upstream cache support interface is explicitly marked as experimental and the non-global client is still usable after all this, so I'm okay with this as is I think.

Reply:
Aha, in that case we actually need to provide a `Close` implementation - pushed that in [`77d8c84` (#6326)](https://github.com/dagger/dagger/pull/6326/commits/77d8c842be73a28da8f4afe60bcde6f204dd5cd8).